sunpy/map/maphelper.py

```diff
@@ -0,0 +1,200 @@
+    reference_pixel :`~astropy.units.Quantity` of size 2, optional
+        Reference pixel along each axis.
+        first index is the x axis, second index is the y axis.
+        Defaults to the center of data array, ``(data.shape[0] + 1)/2., (data.shape[1] + 1)/2.)``.
+    scale : `~astropy.units.Quantity` of size 2, optional
+        Pixel scaling along x and y axis (i.e. the spatial scale of the pixels (dx, dy))
+        Defaults to ``([1., 1.] arcsec)``.
+    **kwargs:
+        Additional arguments that will be put into metadict header. These must be in the
+        list of acceptable meta keywords.
+
+    Returns
+    -------
+    `~sunpy.util.MetaDict`
+        A `sunpy.util.MetaDict` that contains the header information required for making a `sunpy.map.Map`
+
+    Examples
+    --------
+    >>> from sunpy import map
+    >>> from sunpy.coordinates import frames
+    >>> from astropy.coordinates import SkyCoord
+    >>> from astropy import units as u
+    >>> import numpy as np
+
+    >>> data = np.random.rand(1024, 1024)
+    >>> my_coord = SkyCoord(0*u.arcsec, 0*u.arcsec, obstime="2017-08-01", observer = 'earth', frame=frames.Helioprojective)
+    >>> my_header = map.make_header(data, my_coord)
+    >>> my_map = map.Map(data, my_header)
+    """
+    if not isinstance(coordinate, (SkyCoord, frames.BaseCoordinateFrame)):
+        raise ValueError("coordinate needs to be an Astropy coordinate frame or SkyCoord instance.")
+    meta = _get_meta_from_coordinate(coordinate)
+
+    meta['crval1'], meta['crval2'] = coordinate.Tx.value, coordinate.Ty.value
+
+    if reference_pixel == 'center':
+        reference_pixel = u.Quantity([(data.shape[0] + 1)/2.*u.pix, (data.shape[1] + 1)/2.*u.pix])
+
+    meta['crpix1'], meta['crpix2'] = reference_pixel[0], reference_pixel[1]
+
+    meta['cdelt1'], meta['cdelt2'] = scale[0], scale[1]
+
+    meta_dict = MetaDict(meta)
+
+    for key in kwargs:
+        if key in acceptable_meta:
+            meta_dict[key] = kwargs[key]
+
+    return meta_dict
+
+
+def _get_meta_from_coordinate(coordinate):
+    """
+    Function to get WCS meta data from an `astropy.coordinates.SkyCoord` or frame using
+    `astropy.wcs.utils.celestial_frame_to_wcs`.
+
+    Parameters
+    ----------
+    coordinate : `~astropy.coordinates.SkyCoord` or `~astropy.coordinates.BaseFrame`
+        Coordinate object to pull metadata from.
+
+    Returns
+    -------
+    `dict`
+        Containing the meta information required for `sunpy.map.Map` header:
+            * ctype1, ctype2
+            * cunit1, cunit2
+            * hgln_obs, hglt_obs
+            * dsun_obs
+            * rsun_obs
+            * date_obs
+    """
+
+    coord_meta = {}
+
+    if not isinstance(coordinate, (SkyCoord, frames.BaseCoordinateFrame)):
+        raise ValueError("Input must be an Astropy coordinate frame or SkyCoord instance.")
+
+    if isinstance(coordinate, SkyCoord):
+        if coordinate.obstime is None or coordinate.frame is None:
```

Comment:
I don't think it's possible to have a `SkyCoord` without a frame. I suggest you convert a `SkyCoord` to a frame here:
```python
if isinstance(coordinate, SkyCoord):
    coordinate = coordinate.frame
```

and then just implement all the logic after that for frames and not SkyCoord.